Mathlib/RingTheory/Valuation/Basic.lean

```diff
@@ -211,6 +211,9 @@ theorem ne_zero_iff [Nontrivial Γ₀] (v : Valuation K Γ₀) {x : K} : v x ≠
 theorem unit_map_eq (u : Rˣ) : (Units.map (v : R →* Γ₀) u : Γ₀) = v u :=
   rfl
 
+theorem ne_zero_of_unit [Nontrivial Γ₀] (v : Valuation K Γ₀) (x : Kˣ) : v x ≠ (0 : Γ₀) := by
+  simp only [ne_eq, Valuation.zero_iff, Units.ne_zero x, not_false_iff]
```

Comment:
Could you please add a variant that assumes `IsUnit x` for `x : K` (as opposed to a bundled unit)?

Reply:
Done!